### PR TITLE
fix failed unit tests

### PR DESCRIPTION
**Pull request by Starnop**
Signed-off-by: Starnop <starnop@163.com>

### Ⅰ. Describe what this PR did

Previously, CI will looks good even if the unit test did not pass.  So there were a lot of tests that didn't actually pass before. And update them in this pull request. As for fix the bug of CI, please refer to #340 
### Ⅱ. Does this pull request fix one issue?

None.

### CODE REVIEW

File: dfget/config/config.go [hunk 1]
```diff
@@ -276,7 +270,7 @@ func AssertConfig(cfg *Config) (err error) {
 	}
 
 	if err := checkURL(cfg); err != nil {
-		return errors.Wrapf(errType.ErrInvalidValue, "url: %v", err)
+		return errors.Wrapf(errType.ErrInvalidValue, "url %+v", err)
 	}
 
 	if err := checkOutput(cfg); err != nil {
```
Comment (lowzj) [inline dfget/config/config.go]: We should wrap this error. I think that it's also an invalid value error.

File: dfget/config/config_test.go [hunk 2]
```diff
@@ -95,16 +96,14 @@ func (suite *ConfigSuite) TestAssertConfig(c *check.C) {
...
 		checkFunc func(err error) bool
 	}{
 		{checkFunc: errors.IsNotInitialized},
-		{clog: clog, checkFunc: errors.IsNotInitialized},
-		{clog: clog, slog: clog, checkFunc: errors.IsInvalidValue},
-		{clog: clog, slog: clog, url: "http://a.b", checkFunc: errors.IsNilError},
-		{clog: clog, slog: clog, url: "http://a.b", output: "/root", checkFunc: errors.IsInvalidValue},
+		{clog: clog, checkFunc: errors.IsInvalidValue},
+		{clog: clog, url: "http://a", checkFunc: errors.IsInvalidValue},
+		{clog: clog, url: "http://a.b.com", output: "/tmp/output", checkFunc: errors.IsNilError},
```
Comment (lowzj) [inline dfget/config/config_test.go]: We should not change the output `/root`

**Comment by Starnop** [comment]
All mentioned above has been updated. PTAL.

**Comment by lowzj** [comment]
LGTM

**Pull request merged**